src/generators/jsx/utils/transformer.mjs

```diff
@@ -0,0 +1,24 @@
+import { visit } from 'unist-util-visit';
+import { TAG_TRANSFORMS } from '../constants.mjs';
+
+/**
+ * Transforms elements in a syntax tree by replacing tag names according to the mapping.
+ *
+ * @param {Object} tree - The abstract syntax tree to transform
+ * @returns {Object} The transformed tree (modified in place)
+ */
+export default () => tree => {
+  visit(tree, ['raw', 'element'], node => {
+    // TODO(@avivkeller): Our parsers shouldn't return raw nodes
+    // When they mistake "<Type>" for an HTML node, rather than
+    // the string type that it is.
+    if (node.type === 'raw') {
```

Comment:
I'd rather have a switch statement here :P

Reply:
> I'd rather have a switch statement here :P

There's only one clause, so I think we should leave it as is.

> Or simply have `text` within `TYPE_TRANSFORMS`

The transforms we are apply are for `.tagName`, not `.type`

Reply:
I got that, but you are doing a check both for .type and .tagName; So you can have a map for .tagName (TAG_TRANSFORMS) and one for .type (TYPE_TRANSFORMS) in a way that this can be simplified to:

```jsx
node.type = node.type in TYPE_TRANSFORMS ? TYPE_TRANSFORMS[node.type] : node.type;
node.tagName = node.tagName in TAG_TRANSFORMS ? TAG_TRANFORMS[node.tagName] : node.tagName;
```

Reply:
So pretty much your code below can be simplified to the code above and reduce mental strain